cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -21,19 +21,21 @@ type SimulateTransactionCost struct {
 	MemoryBytes     uint64 `json:"memBytes,string"`
 }
 
-type SimulateTransactionResult struct {
-	Auth      []string `json:"auth"`
-	Events    []string `json:"events"`
-	Footprint string   `json:"footprint"`
-	// TODO: update documentation and review field name
-	XDRs []string `json:"xdrs"`
+// SimulateHostFunctionResult contains the simulation result of each HostFunction within each InvokeHostFunctionOp
+type SimulateHostFunctionResult struct {
+	Auth []string `json:"auth"`
+	XDR  string   `json:"xdr"`
 }
 
 type SimulateTransactionResponse struct {
-	Error        string                      `json:"error,omitempty"`
-	Results      []SimulateTransactionResult `json:"results,omitempty"`
-	Cost         SimulateTransactionCost     `json:"cost"`
-	LatestLedger int64                       `json:"latestLedger,string"`
+	Error string `json:"error,omitempty"`
+	// TODO: update documentation and review field names
+	TransactionData string                       `json:"transaction_data"` // SorobanTransactionData XDR in base64
+	Events          []string                     `json:"events"`           // DiagnosticEvent XDR in base64
+	MinResourceFee  int64                        `json:"minResourceFee,string"`
+	Results         []SimulateHostFunctionResult `json:"results,omitempty"`
```

Comment:
for consideration, remove the `omitempty`, to include results array in response, consistent with other collections in the struct like the events. makes a little easier to use from client side, doesn't have to check for absence first, etc.

Reply:
How would it behave with classic transaction ?

Reply:
the `results` array would be empty, clients figure out what do when no results, but at least they can depend on presence and don't have to null check first then reference the length, in cases where you expect this to be absent instead, should those be conveyed more as an `error`?